crates/bevy_ecs/src/entity/mod.rs

```diff
@@ -250,8 +256,8 @@ impl Entities {
         // Use one atomic subtract to grab a range of new IDs. The range might be
         // entirely nonnegative, meaning all IDs come from the freelist, or entirely
         // negative, meaning they are all new IDs to allocate, or a mix of both.
-        let range_end = self.free_cursor.fetch_sub(count as i64, Ordering::Relaxed);
-        let range_start = range_end - count as i64;
+        let range_end = self.free_cursor.fetch_sub(count as Int, Ordering::Relaxed);
```

Comment:
This has wrapping behaviour on overflow, which means that we don't fail gracefully when the `free_cursor` overflows.
On the other hand, that requires `2^31` items to be requested between flushes, which does seem unlikely.
Luckily, we just panic further below at the indexing in that case, so it's not the end of the world.
If you get to ridiculous numbers, like requesting `u32::MAX-k`, I think the subtraction below would also wrap in release, potentially leading to some very strange results.

Reply:
Ok, this is the kind of thing I was unsure of the impact of a change like this. Would the "proper" solution be to drop ID sizes down to `u16` or something to avoid this, or is this wrapping behavior simply a caveat that would need to be documented for these kinds of platforms?

Reply:
I wonder if we can / should fix this in general 🤔 If so, we should likely split it out into its own PR though.

Reply:
For the other platforms, since `i64` is so much bigger than `u32`, if too many entities are requested, we should always panic at the `try_from` below, as far as I can tell.